project-repo/review-process.md

```diff
@@ -2,71 +2,73 @@
-* Automated PR for dependency updates should be looked at with scrutiny: it’s not only because tests are passing that it’s fine to merge them. Look at the upstream changelog/commits, assess the impact, see if we can benefit from the changes and reduce technical debt. If any changes are required to have the tests passing again, those changes should be added to the same branch updating the dependency directly.
+* Any new code and fixes should be accompanied with new or modified existing tests.  Only fixes for rare crashes that are exceptionally difficult to reproduce are acceptable without accompanying automated tests.  If that is the case for your particular fix, please ensure the description reflects this.
+* Any commits merged in the `main` branch need to pass the tests. This is our mainline. You can split your changes into multiple, more discrete modifications, so as to help give additional context when one needs to dig into the git history later on.
+* Reviewing a change is serious business. A generic "LGTM" should be more the exception than the rule. Comments are encouraged and should be seen as positive feedback to broaden our understanding of the issue at hand. Accordingly, the commenter should strive to make their feedback as constructive as possible.
+* When there are multiple reviewers on a PR, please pay attention to the comments and answers of your peers!  This helps building shared understanding of the code. All of us could always learn something new. :-)
+* Try to not let an open pull request remain without activity for more than 3 days. Especially, try to be responsive within a day for contributions from external contributors. Stale branches should be either worked on or deleted. This helps ensure that the repository has an easy and actionable pull request review list.
+* Automated pull requests for dependency updates should be looked at with scrutiny: tests passing does not necessarily imply that the pull request is good to merge. Examine at the upstream changelog and/or commit logs, and assess the impact to see if we would benefit from the changes, and generally aim to reduce technical debt. If any changes are required to make the tests pass again for such a pull request, those changes should be added to the very same branch, so that merging the pull request would not leave the tests broken.
 
 ### Merge workflow
 
-We should use the merge commit workflow (not squash and merge). The history of each branch should make sense and not more "fixup" commits should be there.
+We should use the **merge commit workflow** (not squash and merge). The history of each branch should make sense on its own, with few or no "fixup" commits.
+
+However, for better readability of the mainline, the merge commit in the mainline should follow [conventional commits][convcommits]. This can be enforced [on a per project][gh-docs-merge-commit] basis on GitHub.
```

Comment:
I know the original text was mentioning this conventional commits. I still need to dig if this is what we really want or not (I think we need to change this as I tried to experiment with it and I am not convinced at all). do you mind adding a FIXME here?